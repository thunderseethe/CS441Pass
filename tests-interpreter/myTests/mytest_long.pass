//Test for the new long type

begin
	int n =  2147483647;
	int m =  2147483647;
	print (n + m); //Should over flow because they're ints
	//Testing that number literals are properly stored as longs if they're over INT_MAX
	print (2147483647 + 2147483647);
	long j = 2147483647; //Test storing longs
	print (j * 3); // Test type coercion for longs
end 