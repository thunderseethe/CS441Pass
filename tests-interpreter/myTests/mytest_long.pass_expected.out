-2
4294967294
6442450941
